bar_app/views/bars/bar_view.py

```diff
@@ -0,0 +1,70 @@
+from rest_framework import status, generics, filters
+from rest_framework.response import Response
+from rest_framework.permissions import IsAdminUser, IsAuthenticated
+from bar_app.models import Bar
+from bar_app.serializers import BarSerializer
+
+class BarListView(generics.ListAPIView):
+    """
+    Retrieve a list of all bars.
+    """
+    queryset = Bar.objects.all()
+    serializer_class = BarSerializer
+    permission_classes = [IsAdminUser, IsAuthenticated]
+    filter_backends = [filters.OrderingFilter, filters.SearchFilter]
+    search_fields = ["id", "name"]
+    ordering_fields = ["id", "name"]
+
+class BarCreateView(generics.CreateAPIView):
+    """
+    Create a new bar.
+    """
+    queryset = Bar.objects.all()
+    serializer_class = BarSerializer
+    permission_classes = [IsAdminUser, IsAuthenticated]
+
+    def create(self, request, *args, **kwargs):
+        if isinstance(request.data, dict):
+            serializer = self.get_serializer(data=request.data)
+        elif isinstance(request.data, list):
+            serializer = self.get_serializer(data=request.data, many=True)
```

Comment:
what is this piece of code supposed to handle ? You can create batches of bars ? 😕

Reply:
Yes it is, create one bar or multiple bars (same for reference). This could be an option maybe?

Reply:
it changes quite a lot the initial API asked. but keep it as is